### PR TITLE
Added check for non-procedural IBL support to IBL sandcastle

### LINKED ISSUE #8230: IBL Sandcastle demo issues
http://localhost:8080/Apps/Sandcastle/index.html?src=Image-Based%20Lighting.html&label=All

1. Under Safari/Mac, the `Use procedural image` check does nothing. If it's not supported on some systems, we should be indicating that.
2. On Chrome/Linux, everything works but the `Use procedural image` checkbox seems to be flipped, i.e. when it's off it uses an image and when it's on it doesn't:

image

CC @lilleyse @IanLilleyT

### PR DESCRIPTION
**Pull request by IanLilleyT**
Fixes #8230

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @IanLilleyT!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/IBL_Sandcastle_Fix/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

File: Apps/Sandcastle/gallery/Image-Based Lighting.html [hunk 1]
```diff
@@ -107,15 +107,19 @@
...
-    Sandcastle.addToggleButton('Use procedural image', false, function(checked) {
-        if (!checked) {
-            model.sphericalHarmonicCoefficients = coefficients;
-            model.specularEnvironmentMaps = environmentMapURL;
-        } else {
-            model.sphericalHarmonicCoefficients = undefined;
-            model.specularEnvironmentMaps = undefined;
-        }
-    });
+    if (Cesium.OctahedralProjectedCubeMap.isSupported(viewer.scene.context)) {
```
Comment (lilleyse) [inline Apps/Sandcastle/gallery/Image-Based Lighting.html]: `OctahedralProjectedCubeMap` is private and so is `scene.context`. You'll need to add a new supported check, probably to `Scene` - `scene.specularEnvironmentMapsSupported`

File: Apps/Sandcastle/gallery/Image-Based Lighting.html [hunk 2]
```diff
@@ -107,15 +107,19 @@
...
+            if (!checked) {
+                model.sphericalHarmonicCoefficients = coefficients;
+                model.specularEnvironmentMaps = environmentMapURL;
+            } else {
+                model.sphericalHarmonicCoefficients = undefined;
+                model.specularEnvironmentMaps = undefined;
+            }
+        });
+    } else {
+        console.log("This browser doesn't support texture environment maps.");
```
Comment (lilleyse) [inline Apps/Sandcastle/gallery/Image-Based Lighting.html]:
```suggestion
        console.log('This browser doesn't support specular environment maps.');
```

Use single-quotes for strings. Also `texture environment maps` is a bit redundant so I replaced `texture` with `specular` which is what the option is called.

**Comment by IanLilleyT** [comment]
ready @lilleyse

File: Source/Scene/Scene.js [hunk 3]
```diff
@@ -948,6 +948,18 @@ define([
             }
         },
 
+        /**
+         * Returns <code>true</code> if specular environment maps are supported.
+         * @memberof Scene.prototype
+         * @type {Boolean}
+         * @default true
+         */
+        specularEnvironmentMapSupported : {
```
Comment (lilleyse) [inline Source/Scene/Scene.js]: `specularEnvironmentMapsSupported` sounds better

Comment (lilleyse) [reply Source/Scene/Scene.js]: Also model the doc after `invertClassificationSupported`: there shouldn't be a default value and add a `@see` to `specularEnvironmentMaps`

**Comment by IanLilleyT** [comment]
good to go @lilleyse

**Comment by lilleyse** [comment]
CI failed because of https://github.com/AnalyticalGraphicsInc/cesium/issues/8260

**Pull request merged**